src/js/sync.js

```diff
@@ -0,0 +1,89 @@
+var SailthruGigya = {
+
+  sailthru_response : '',
+
+  syncProfile : function(eventObj, callback_url) {
+
+    // If callback enabled send to a server side when requiring User API calls
+    if (callback_url.length > 0) {      
+
+      var x = window.XMLHttpRequest ? new XMLHttpRequest() : new ActiveXObject("Microsoft.XMLHTTP");
+      var profile = SailthruGigya.serialize_profile(eventObj);
+      var params = 'json='+profile;
+
+      x.open("POST", callback_url, true);
+      x.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
+      x.setRequestHeader("Content-length", params.length);
+      x.setRequestHeader("Connection", "close");
+
+      x.onreadystatechange=function()
+      {
+      if (x.readyState==4 && x.status==200)
+        {
+          sailthru_response = x.responseText;
+        } else {
+          sailthru_response = x.readyState;
+        }
+      }
+      x.send(params);
+
+    } else {
+        
+      if (window.Sailthru) {
+        
+        if ( eventObj.user.email.length <=0 || eventObj.user.email !== undefined ) {
+          var vars = {};
+          var lists = {};
+          // default vars to exclude
+          var exclude = 'UIDSig, UIDSignature, signatureTimestamp, capabilities, statusCode, statusReason, signatureTimestamp, isTempUser, isConnected, isLoggedIn, isSiteUID, isSiteUser, oldestDataUpdatedTimestamp';
+          
+          for (let key in eventObj.user) {        
+            
+            if (exclude.indexOf(key) < 0) {
+              if (eventObj.user[key].length > 0) {
+                vars[key] = eventObj.user[key];
+              }
+            }
+          }
+
+          // lists
+          if (SailthruGigya.lists !== undefined) {
+            var lists = SailthruGigya.lists;
```

Comment:
declaring var there scopes it to that conditional.

Reply:
This seems like you can just use SailthruGigya.lists and a null-safe check.

```js
if (SailthruGigya.lists == undefined) {
   SailthruGigya.lists = {};
}
```

Also I'm unsure of the usage of undefined.